src/startupscriptgenerator/src/common/outputHelper.go

```diff
@@ -23,27 +22,25 @@ func AppendScriptToExtractZippedOutput(scriptBuilder *strings.Builder, appDir st
 	}
 
 	if FileExists(fullZipFilePath) {
-		scriptBuilder.WriteString("\n")
-		scriptBuilder.WriteString("echo Extracting '" + consts.CompressedOutputFileName + "' contents...\n")
-		scriptBuilder.WriteString("cd \"" + intermediateDir + "\"\n")
-		scriptBuilder.WriteString("tar -xzf \"" + consts.CompressedOutputFileName + "\"\n")
-		scriptBuilder.WriteString("echo Done.\n")
+		scriptBuilder.AppendEmptyLine()
+		scriptBuilder.Echo("Extracting '" + consts.CompressedOutputFileName + "' contents...")
+		scriptBuilder.ChangeDirectory(intermediateDir)
+		scriptBuilder.ExtractCompressedFile(consts.CompressedOutputFileName)
+		scriptBuilder.Echo("Done")
 	} else {
 		fmt.Printf("Could not find the compressed file '%s'\n", fullZipFilePath)
 	}
 }
 
-func AppendScriptToCopyToDir(scriptBuilder *strings.Builder, srcDir string, destDir string) {
-	scriptBuilder.WriteString("\n")
-	scriptBuilder.WriteString("declare -r srcDir=\"" + srcDir + "\"\n")
-	scriptBuilder.WriteString("declare -r destDir=\"" + destDir + "\"\n")
-	scriptBuilder.WriteString("if [ -d \"$destDir\" ]; then\n")
-	scriptBuilder.WriteString("    echo \"Directory '$destDir' already exists. Deleting it...\"\n")
-	scriptBuilder.WriteString("    rm -rf \"$destDir\"\n")
-	scriptBuilder.WriteString("    echo Done.\n")
-	scriptBuilder.WriteString("fi\n")
-	scriptBuilder.WriteString("mkdir -p \"$destDir\"\n")
-	scriptBuilder.WriteString("echo \"Copying content from '$srcDir' to directory '$destDir'...\"\n")
-	scriptBuilder.WriteString("cp -rf \"$srcDir\"/* \"$destDir\"\n")
-	scriptBuilder.WriteString("echo Done.\n")
+func AppendScriptToCopyToDir(scriptBuilder *ScriptBuilder, srcDir string, destDir string) {
+	scriptBuilder.AppendEmptyLine()
+	scriptBuilder.AppendLine("if [ -d \"" + destDir + "\" ]; then")
```

Comment:
Few complexities related to blocks like `if`:
1. Expressing conditions: condition1 or condition2, condition1 and condition 2
2. Indentation of statements below this if condition. Ideally we might want to use api like `ChangeToDirectory` under this condition too, but the problems are: a). we would loose indentation when generating the script b). Even if we can fix the indentation (keep track of these kind of blocks using a stack) in the generated script, when reading this code in our code base, have indentation in golang code is also needed

Reply:
one thing we could do is to add a `scriptBuilder.AddIfBlock(condition: string)` that returns a `ifBlock` object, in a fluent design. We could do this:
```
scriptBuilder.AddIfBlock("-d \"" + destDir + "\"")
    .appendLine(...)
    .end()
```

Reply:
The aforementioned IfBlock object itself could probably be another `ScriptBuilder` instance